src/d3a/models/strategy/predefined_load.py

```diff
@@ -126,3 +129,56 @@ def area_reconfigure_event(self, **kwargs):
         self._area_reconfigure_prices(**kwargs)
         if key_in_dict_and_not_none(kwargs, 'daily_load_profile'):
             self._event_activate_energy(kwargs['daily_load_profile'])
+
+
+class LoadForecastStrategy(DefinedLoadStrategy):
+    """
+        Strategy responsible for reading single forecast consumption data via hardware API
+    """
+    parameters = ('daily_load_profile', 'fit_to_limit', 'energy_rate_increase_per_update',
+                  'update_interval', 'initial_buying_rate', 'final_buying_rate',
+                  'balancing_energy_ratio', 'use_market_maker_rate')
+
+    def __init__(self, power_forecast_W: float = 0,
+                 fit_to_limit=True, energy_rate_increase_per_update=None,
+                 update_interval=None,
+                 initial_buying_rate: Union[float, dict, str] =
+                 ConstSettings.LoadSettings.BUYING_RATE_RANGE.initial,
+                 final_buying_rate: Union[float, dict, str] =
+                 ConstSettings.LoadSettings.BUYING_RATE_RANGE.final,
+                 balancing_energy_ratio: tuple =
+                 (ConstSettings.BalancingSettings.OFFER_DEMAND_RATIO,
+                  ConstSettings.BalancingSettings.OFFER_SUPPLY_RATIO),
+                 use_market_maker_rate: bool = False):
+        """
+        Constructor of LoadForecastStrategy
+        :param power_forecast_W: forecast for the next market slot
+        """
+        if update_interval is None:
+            update_interval = \
+                duration(minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL)
+
+        super().__init__(daily_load_profile=None,
+                         fit_to_limit=fit_to_limit,
+                         energy_rate_increase_per_update=energy_rate_increase_per_update,
+                         update_interval=update_interval,
+                         final_buying_rate=final_buying_rate,
+                         initial_buying_rate=initial_buying_rate,
+                         balancing_energy_ratio=balancing_energy_ratio,
+                         use_market_maker_rate=use_market_maker_rate)
+        self.power_forecast_buffer_W = power_forecast_W
+
+    def event_activate_energy(self):
+        self.update_energy_forecast()
+
+    def update_energy_forecast(self):
+        self._initiate_hrs_per_day()
+
+        # sets energy forecast for next_market
+        energy_forecast_Wh = (self.area.config.slot_length / duration(hours=1)) * \
+            self.power_forecast_buffer_W
+        assert energy_forecast_Wh >= 0.0
```

Comment:
```suggestion
        assert energy_forecast_Wh >= 0.0, energy_forecast_Wh cannot be a negative value.
```
Minor cosmetic.

Reply:
On second thought, why not raise an exception, so that simulation doesn't crash due to external clients' wrong value ?
```suggestion
        if energy_forecast_Wh <= 0.0:
            raise MarketException("energy_forecast_Wh cannot be a negative value.")
```

Reply:
Its a matter of taste really. 
1. An AssertionError behaves the same as an Exception AFAIK
2. The user can not set a negative value, please have a look into the `_set_power_forecast_impl`, there it is already checked (and does not make the simulation crash because it is caught by the try/except.

TLDR; will raise an Exception as you suggested.

Reply:
You can remove it at all from here as it is already covered in `_set_power_forecast_impl `.

Reply:
yeah, i wanted to keep it as extra layer of security (in case something goes wrong in the conversion between power and energy) But you are right, its not needed as there is just a factor applied.